### PR TITLE
LinkedListIndexed

### PR DESCRIPTION
**Pull request by WareditorGit**
A module that turns a class into a collection of linked lists where each list is composed of instances of the same index.

This will be useful for closureFrames and closureKeyPresses.

Not sure about the name.

### CODE REVIEW

**Comment by Frotty** [comment]
> This will be useful for closureFrames and closureKeyPresses.

It should also be able to be used in ClosureEvents right?

**Comment by WareditorGit** [comment]
Yeah and others if needed

File: wurst/data/LinkedListIndexedModule.wurst [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package LinkedListIndexedModule
+import HashMap
+
+/** Turns a class into a collection of linked lists where each list
+    is composed of instances of the same index. Each instance 
+    knows it's previous and next member in its index's list. */
```
Comment (Cokemonkey11) [inline wurst/data/LinkedListIndexedModule.wurst]: Doesn't answer "Why?". What's this package useful for?

File: wurst/data/LinkedListIndexedModule.wurst [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        function next() returns LinkedListIndexedModule.thistype
+            let res = current
+            current = current.prev
+            return res
+
+        function reset()
+            current = last
+
+        function close()
+            destroy this
```
Comment (Cokemonkey11) [inline wurst/data/LinkedListIndexedModule.wurst]: missing eof lf

File: wurst/data/LinkedListIndexedModule.wurst (hunk 1 above)
Comment (Frotty) [reply wurst/data/LinkedListIndexedModule.wurst]: To modularize stuff we are already doing in the stdlib.

**Comment by Frotty** [comment]
@WareditorGit can you fix eof lf? should adhere to editorconfig.
Did you have any further ideas about other names?

**Comment by WareditorGit** [comment]
Sorry, I am away from my comp during this period. Will work on this when I get back.

**Comment by Frotty** [comment]
@WareditorGit I could also merge like this or are there changes u want to make besides name?

**Comment by WareditorGit** [comment]
No changes.
However I agree that it's might be too peculiar to include in the std lib and might add some unrequired abstraction.

**Comment by Frotty** [comment]
What's the point of not having it in stdlib if we plan to use it in stdlib?

**Comment by WareditorGit** [comment]
I am saying maybe not use it if it's only for those few cases.

**Comment by Frotty** [comment]
Will this work for destroying listeners during a callback without temp var?
 I still want to merge tbh.

**Comment by Frotty** [comment]
bump

**Comment by Cokemonkey11** [comment]
> What's the point of not having it in stdlib if we plan to use it in stdlib?

If you plan to use in stdlib then it should be in stdlib.

It would be interesting to see a concrete stdlib change - in this PR - that moves old code to using this new feature.

**Comment by Frotty** [comment]
I actually like this, but the hashmaps seem overkill, why not just use module attribute. Might fix and merge.

**Pull request closed without merging**